### PR TITLE
Move NSManagedObjectContext  extension from SE

### PR DESCRIPTION
**Pull request by KaterinaWire**
## What's new in this PR?

This PR is a part of the Notification Engine setup. 
`NSManagedObjectContext` extension and `UpdateEventProcessor`  protocol were copied from SE.

## Dependencies

wireapp/wire-ios-sync-engine/pull/1246

### CODE REVIEW

File: Source/Utilis/ZMUpdateEvent.swift [hunk 1]
```diff
@@ -18,6 +18,14 @@
 
 import Foundation
 
+@objc
+public protocol UpdateEventProcessor: class {
```
Comment (typfel) [inline Source/Utilis/ZMUpdateEvent.swift:22]: I think protocol didn't have to move to data model, it could stay in request-strategy?

**Comment by typfel** [comment]
Consider also merging this into a feature branch.

File: Source/Utilis/ZMUpdateEvent.swift (hunk 1 above)
Comment (KaterinaWire) [reply Source/Utilis/ZMUpdateEvent.swift]: Ok.
Done

**Comment by KaterinaWire** [comment]
> Consider also merging this into a feature branch.

The base branch was changed.

**Pull request merged**